game/quake2_md2.ksy

```diff
@@ -0,0 +1,245 @@
+    contents: IDP2
+  - id: version
+    type: u4
+    doc: Always 8, apparently.
+  - id: skin_width_px
+    type: u4
+  - id: skin_height_px
+    type: u4
+  - id: bytes_per_frame
+    type: u4
+  - id: num_skins
+    type: u4
+  - id: vertices_per_frame
+    type: u4
+  - id: num_tex_coords
+    type: u4
+  - id: num_triangles
+    type: u4
+  - id: num_gl_cmds
+    type: u4
+  - id: num_frames
+    type: u4
+  - id: ofs_skins
+    type: u4
+  - id: ofs_tex_coords
+    type: u4
+  - id: ofs_triangles
+    type: u4
+  - id: ofs_frames
+    type: u4
+  - id: ofs_gl_cmds
+    type: u4
+  - id: ofs_eof
+    type: u4
+instances:
+  skins:
+    pos: ofs_skins
+    size: 64
+    type: strz
+    encoding: ascii
+    repeat: expr
+    repeat-expr: num_skins
+  tex_coords:
+    pos: ofs_tex_coords
+    type: tex_point
+    repeat: expr
+    repeat-expr: num_tex_coords
+  triangles:
+    pos: ofs_triangles
+    type: triangle
+    repeat: expr
+    repeat-expr: num_triangles
+  frames:
+    pos: ofs_frames
+    type: frame
+    repeat: expr
+    repeat-expr: num_frames
+  gl_cmds:
+    pos: ofs_gl_cmds
+    type: gl_cmd
+    repeat: until
+    repeat-until: _.cmd_num_vertices == 0
+  anim_names:
+    value: |
+      ['stand', 'run', 'attack', 'pain1', 'pain2', 'pain3', 'jump', 'flip',
+      'salute', 'taunt', 'wave', 'point', 'crstnd', 'crwalk', 'crattak',
+      'crpain', 'crdeath', 'death1', 'death2', 'death3']
+  anim_start_indexes:
+    value: |
+      [0, 40, 46, 54, 58, 62, 66, 72, 84, 95, 112,
+      123, 135, 154, 160, 169, 173, 178, 184, 190]
+  anim_num_frames:
+    value: |
+      [40, 6, 8, 4, 4, 4, 6, 12, 11, 17, 11, 12, 19, 6, 9, 4, 5, 6, 6, 8]
+types:
+  tex_point:
+    seq:
+      - id: u_px
+        type: u2
+      - id: v_px
+        type: u2
+    instances:
+      u_normalized:
+        value: u_px.as<f4> / _root.skin_width_px
+      v_normalized:
+        value: v_px.as<f4> / _root.skin_height_px
+  triangle:
+    seq:
+      - id: vertex_indices
+        type: u2
+        repeat: expr
+        repeat-expr: 3
+      - id: tex_point_indices
+        type: u2
+        repeat: expr
+        repeat-expr: 3
+  vec3f:
+    seq:
+      - id: x
+        type: f4
+      - id: y
+        type: f4
+      - id: z
+        type: f4
+  compressed_vec:
+    seq:
+      - id: x_compressed
+        type: u1
+      - id: y_compressed
+        type: u1
+      - id: z_compressed
+        type: u1
+    instances:
+      x:
+        value: x_compressed * _parent._parent.scale.x + _parent._parent.translate.x
+      y:
+        value: y_compressed * _parent._parent.scale.y + _parent._parent.translate.y
+      z:
+        value: z_compressed * _parent._parent.scale.z + _parent._parent.translate.z
+  vertex:
+    seq:
+      - id: position
+        type: compressed_vec
+      - id: normal_index
+        type: u1
+        doc: `normal = bytedirs[normal_index]`
+        doc-ref: |
+          https://github.com/skullernet/q2pro/blob/master/src/common/math.c#L80
```

Comment:
Referring to ever-changing references (branches) like `master` is not future-proof. Even more so if you link to a specific line number (which is a good thing, and it should be done whenever possible) - in general, there's a very good chance that soon there will be at least something quite different at line 80 than it is now. In the long run (several years) it's also not uncommon that code gets restructured/moved and the referenced source file will no longer exist on the `master` branch.

Therefore, you should use something permanent like a version tag (if available) or a commit hash (I like it shortened, because that's usually enough to identify the commit and doesn't disturb much), for example:

```suggestion
          https://github.com/skullernet/q2pro/blob/f4faabd/src/common/math.c#L80
```

Reply:
Aha, very good point. I will do this in future.